src/main/scala/viper/gobra/frontend/info/implementation/resolution/NameResolution.scala

```diff
@@ -284,8 +289,31 @@ trait NameResolution { this: TypeInfoImpl =>
 
       case n =>
         (n, lookup(sequentialDefenv(n), serialize(n), UnknownEntity())) match {
-          // in case no entity was found in the current package, look for it in unqualifiedly imported packages:
-          case (n: PIdnUse, UnknownEntity()) => tryUnqualifiedPackageLookup(n)
+
+          // lookup has failed
+          case (n: PIdnUse, UnknownEntity()) =>
+            // in case no entity was found in the current package, look for it in
+            // - unqualifiedly imported packages or
+            // - the current interface if `n` occurs in an interface definition
+            (tryUnqualifiedPackageLookup(n), tryEnclosingInterface(n)) match {
+              // no entity was found in an unqualifiedly imported packages but we are inside an interface definition
+              case (UnknownEntity(), Some(it)) =>
+                // `n` appears in an interface and due to the way Go works, interface definitions (i.e. methods & predicates)
+                // have not been considered so far. Therefore, we perform a second-level lookup just on the definitions that
+                // the interface provides
+                val interfaceEntities = definitionsForScope(it).map(id => (serialize(id), defEntity(id)))
+                // create an environment only consisting of the definitions and their corresponding entities that the
+                // interface provides:
+                val specialEnv = rootenv(interfaceEntities:_*)
```

Comment:
you should outline the lookup table creation to an attribute,  i.e. something along thr lines of:

```
lazy val interfaceDefenv: PInterface ==>  Environment =
 attr[PInterface, Environment] { itf =>
   rootenf(definitionsForScope(it).map(id => (serialize(id), defEntity(id))) :_*)
 }
```

You could even add add a new trait ans abstract over PInterface.